llvm/test/CodeGen/X86/fold-loop-of-urem.ll

```diff
@@ -0,0 +1,884 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -mtriple=x86_64-unknown-unknown | FileCheck %s
```

Comment:
It would be better to convert this into an opt test.
```suggestion
; RUN: opt -codegenprepare -S < %s | FileCheck %s
```

Reply:
Need the triple (for CGP), but making it an opt test.

Reply:
Actually, not sure why, but most of the tests fail w/ opt. Not sure why that is....

Reply:
Are you using `-passes=codegenprepare`? IIRC CGP hasn't been ported to new pass manager.

Reply:
Okay, the issue is that just `opt -codegenprepare` doesn't actually run the loop analysis passes which this optimization requires. I can get it working if I do something like add `-licm`. Do you know the arguments to run the loop analysis pass as a pre-req?

Reply:
https://github.com/llvm/llvm-project/blob/97a2bd8415dc6792b99ec0f091ad7570673c3f37/llvm/lib/CodeGen/CodeGenPrepare.cpp#L566

IIRC it always requires LI. It is weird.

Reply:
llc runs `LoopSimplify` before `CodeGenPrepare`, which introduces pre-headers for loops in `InsertPreheaderForLoop`.

See also https://github.com/llvm/llvm-project/blob/f0eb5587ceeb641445b64cb264c822b4751de04a/llvm/test/CodeGen/RISCV/O3-pipeline.ll#L42-L73

`Canonicalize natural loops` stands for `LoopSimplify`.

Reply:
Okay, ill try loopsimplify + codegenprepare passes for the `opt` tests.